src/adaptive_graph_of_thoughts/domain/services/neo4j_utils.py

```
@@ -1,216 +1,57 @@
-        logger.info("Closing Neo4j driver.")
+    if _driver is not None:
         _driver.close()
         _driver = None
-    else:
-        logger.info("Neo4j driver is already closed or not initialized.")
 
-# --- Query Execution ---
+
 async def execute_query(
     query: str,
     parameters: Optional[Dict[str, Any]] = None,
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)